### PR TITLE
fix: use `target` from webExtConfig if `browser` option is undefined

### PR DESCRIPTION
**Pull request by aleclarson**
Based on #231 so merge that first.

If the `browser` plugin option is undefined, check the `webExtConfig.target` option before defaulting to chromium.

### CODE REVIEW

File: packages/vite-plugin-web-extension/src/extension-runner/web-ext-runner.ts [hunk 1]
```diff
@@ -27,9 +27,12 @@ export function createWebExtRunner(
 
       const initialConfig = await loadConfig({ pluginOptions, logger, paths });
       const target =
-        pluginOptions.browser === null || pluginOptions.browser === "firefox"
-          ? null
-          : "chromium";
+        pluginOptions.browser === "chrome"
+          ? "chromium"
+          : pluginOptions.browser === null ||
```
Comment (aleclarson) [inline packages/vite-plugin-web-extension/src/extension-runner/web-ext-runner.ts:32]: I wonder if we should deprecate the `browser: null` setting. It's a bit cryptic 😅

File: packages/vite-plugin-web-extension/src/options.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ export interface UserOptions {
    *
    * @default "chrome"
    */
-  browser?: string;
+  browser?: 'chrome' | 'firefox' | null;
```
Comment (aklinker1) [inline packages/vite-plugin-web-extension/src/options.ts:65]: This should remain a `string`. You can target more than just chrome or firefox. Safari, opera, etc.

File: packages/vite-plugin-web-extension/src/options.ts [hunk 3]
```diff
@@ -119,7 +119,7 @@ export interface ResolvedOptions {
   additionalInputs: string[];
   disableAutoLaunch: boolean;
   watchFilePaths: string[];
-  browser?: string;
+  browser?: 'chrome' | 'firefox' | null;
```
Comment (aklinker1) [inline packages/vite-plugin-web-extension/src/options.ts:122]: Same

File: packages/vite-plugin-web-extension/src/options.ts (hunk 2 above)
Comment (aleclarson) [reply packages/vite-plugin-web-extension/src/options.ts:65]: Oops! That's strictly for the manifest templating? I don't see any handling for Safari and others in `createWebExtRunner`, hence the assumption. 😄

File: packages/vite-plugin-web-extension/src/options.ts [hunk 4]
```diff
@@ -62,7 +62,7 @@ export interface UserOptions {
    *
    * @default "chrome"
    */
-  browser?: string;
+  browser?: 'chrome' | 'firefox' | (string & {}) | null;
```
Comment (aleclarson) [inline packages/vite-plugin-web-extension/src/options.ts:65]: this is a TypeScript trick that provides auto-completion for some values while still allowing any string

Comment (aklinker1) [reply packages/vite-plugin-web-extension/src/options.ts:65]: Interesting, didn't know this worked... I've only used branded types with a property inside the `{}`. I've always had to use a function overload to get this behavior before, very good to know!

File: packages/vite-plugin-web-extension/src/extension-runner/web-ext-runner.ts (hunk 1 above)
Comment (aklinker1) [reply packages/vite-plugin-web-extension/src/extension-runner/web-ext-runner.ts:32]: Huh, yeah, wonder what I was thinking lol. Let's leave it for now.

**Review by aklinker1 — APPROVED** [review]

**Pull request merged**